src/realm/sync/noinst/sync_metadata_schema.cpp

```diff
@@ -283,8 +280,6 @@ SyncMetadataSchemaVersions::SyncMetadataSchemaVersions(const TransactionRef& tr)
     // Commit and revert to the original transact stage
     if (orig == DB::transact_Reading)
         tr->commit_and_continue_as_read();
-    else
-        tr->commit_and_continue_writing();
```

Comment:
I couldn't find anywhere that this was being used where this commit would do anything useful and it made the client reset tests which verified that exactly two commits were made more complicated.